### PR TITLE
docs: add a few more docs to ControlFlow

### PR DESCRIPTION
**Pull request by mattsse**
also renames function to `block_number`

### CODE REVIEW

File: crates/stages/src/pipeline/ctrl.rs [hunk 1]
```diff
@@ -10,14 +12,20 @@ pub enum ControlFlow {
         /// The block that caused the unwind.
         bad_block: SealedHeader,
     },
-    /// The pipeline is allowed to continue executing stages.
+    /// The pipeline made progress.
     Continue {
-        /// Block number reached by the stage.
+        /// Block number reached by the last stage in the pipeline.
```
Comment (onbjerg) [inline crates/stages/src/pipeline/ctrl.rs:17]: It's not the last stage, this enum is also used in the pipeline itself per stage. The block number corresponds to the block number reached by the stage the control flow event is for

Comment (onbjerg) [reply crates/stages/src/pipeline/ctrl.rs:17]: yes

**Comment by mattsse** [comment]
@onbjerg updated and added more info to `run_loop`

**Review by onbjerg — APPROVED** [review]
soz for slow approval on this

**Comment by onbjerg** [comment]
@mattsse, tests do not pass on top of latest main, see merge queue kick reason

**Pull request merged**